core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1847,21 +1847,22 @@ class KafkaController(val config: KafkaConfig,
       // Determine which partitions we will accept the new ISR for
       val adjustedIsrs: Map[TopicPartition, LeaderAndIsr] = isrsToAlter.flatMap {
         case (tp: TopicPartition, newLeaderAndIsr: LeaderAndIsr) =>
-          val partitionError: Errors = controllerContext.partitionLeadershipInfo(tp) match {
+          controllerContext.partitionLeadershipInfo(tp) match {
             case Some(leaderIsrAndControllerEpoch) =>
               val currentLeaderAndIsr = leaderIsrAndControllerEpoch.leaderAndIsr
               if (newLeaderAndIsr.leaderEpoch < currentLeaderAndIsr.leaderEpoch) {
-                Errors.FENCED_LEADER_EPOCH
+                partitionResponses(tp) = Left(Errors.FENCED_LEADER_EPOCH)
+                None
+              } else if (newLeaderAndIsr.equalsIgnoreZk(currentLeaderAndIsr)) {
+                // If a partition is already in the desired state, just return it
```

Comment:
It might be worth mentioning that this could happen in the case of a retry after a successful update.